waiter/src/waiter/process_request.clj

```diff
@@ -408,13 +409,23 @@
   (track-process-error-metrics descriptor)
   (utils/exception->response exception request))
 
+(defn cleanup-response-headers
+  "Removes connection-specific header fields when transmitting HTTP/1.x message via HTTP/2."
+  [{:keys [headers] :as response} internal-protocol backend-protocol]
+  (cond-> response
+    (and (= internal-protocol "HTTP/2.0") (not= backend-protocol "HTTP/2.0"))
```

Comment:
Shouldn't we be removing these headers in all cases, not use HTTP/2?  See https://www.mnot.net/blog/2011/07/11/what_proxies_must_do "Remove Hop-by-hop headers"

Reply:
We do this already for the request headers.

Reply:
I had missed that hop-by-hop headers need to be removed for responses too. Thanks, done.